File: rb/features.rb
# SodeomSearch SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module SodeomSearchFeatures
  def self.make_feature(name)
    case name
    when "base"
      SodeomSearchBaseFeature.new
    when "test"
      SodeomSearchTestFeature.new
    else
      SodeomSearchBaseFeature.new
    end
  end
end

File: py/features.py
# SodeomSearch SDK feature factory

from feature.base_feature import SodeomSearchBaseFeature
from feature.test_feature import SodeomSearchTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SodeomSearchBaseFeature(),
        "test": lambda: SodeomSearchTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
